### ISSUE TITLE
Some queries regarding the model design

### ISSUE THREAD

**Issue by Zhang-Y-J** [huanghoujing/person-reid-triplet-loss-baseline#15]
Hi! Great work!
I am trying to understand your model and implemented it in keras myself. There are few areas which I am uncertain of so could u give me some enlightenments?

1. For the base model Resnet-50, other than changing the stride to 1 and removing the top layer, is there any other changes you have made on it? Besides, I notice that for your model, unlike the paper, you have 
    i) x = F.avg_pool2d(x, x.size()[2:])
    ii) x = x.view(x.size(0), -1)
after coming out from Resnet. May I know what are some of the reasons you chose to do these two layers, instead of following the paper, which suggest 2 dense layer and a batchnorm in between? In your case, what would be the output shape of your last flatten layer?

2. It seems that you did not use the default dataset split provided by market1501. What are some of potential advantages using your split? In addition, why there are only 100 persons in the validation set? Shouldnt it be the same number of classes in the training set?

3. I noticed that you also did preprocessing before passing images to the model, however, the values for your image mean ([0.486, 0.459, 0.408]) are different from that provided in imagenet.preprocess ([0.485, 0.456, 0.406]) specified in keras source code. May I know what is the reason for the change?

4. In short, my keras implementation did not perform well on market1501, for the map and top1 top5 are quite low. Is it possible to give me some suggestions (anything) so that I can try to improve my model?

Thank you and hope to hear from you soon!

**Comment by huanghoujing** [comment]
Hi, thanks for your attention.

1.1. No other changes are made.

1.2. I find that vanilla average pooling is really well, so I did not add additional embedding layer after ResNet-50 in the released code. I also did experiments with one additional embedding layer, the result of which I can not remember now. So I also did not provide results for it.

1.3. The output feature shape is `[N, 2048]`, N being the batch size.

2.1. I use the standard protocol of Market1501.

2.2. Choice of validation set is at your will, it's just to (1) make sure the training does not go wrong and (2) to select some super parameters. Generally, validation set should be images or persons that does not appear in the training set.

3. The difference between [0.486, 0.459, 0.408] and [0.485, 0.456, 0.406] is trivial, maybe because of the difference between 1/255 and 1/256, which I can't remember quite well now.

4. You can look into your training log, e.g. the proportion of triplets that satisfies `margin > 0` and `margin > your_margin` (e.g. 0.3) during training. It may be due to the failure of training for some training settings, or potentially due to some mistakes.

**Comment by Zhang-Y-J** [comment]
Hi! Thanks for your reply!
We managed to find out that the low result was due to our wrong labelling. Currently we obtain 84% top 1 and 65% map for Market1501 single-query stride 1 follow the standard market1501 protocol (exclude same camera id), without re-ranking. We were trying to replicate your model however our performance is much lower than yours. Apart from the fact that our gpu only support 32 * 3 within one batch. the models should be the same. I am wondering why my model cannot get similar results to your model. Is it possible for you to provide some enlightenments to us, such as small things you did during training/test? Any suggestions would be good! Thank you very much!

**Comment by huanghoujing** [comment]
Hi, I did not tune other things not mentioned in the code. Besides keeping your setting the same as [these several points](https://github.com/huanghoujing/person-reid-triplet-loss-baseline#current-results), you also need to train on `trainval` set to obtain the reported scores.
